src/components/dropdown/dropdown.tsx

```diff
@@ -438,28 +438,27 @@ export class Dropdown implements InteractiveComponent {
   }
 
   private keyDownHandler = (e: KeyboardEvent): void => {
-    const target = e.target as HTMLSlotElement;
+    const target = e.target as HTMLElement;
+
+    if (target !== this.referenceEl) {
+      return;
+    }
+
     const key = e.key;
-    if (
-      this.triggers.includes(target) ||
-      this.triggers.some((trigger) => trigger.contains(target))
-    ) {
-      const ignoredNodeTypes = ["BUTTON", "CALCITE-BUTTON", "CALCITE-ACTION"];
-      if (ignoredNodeTypes.includes(target.nodeName)) {
-        if (this.active && (key === "Escape" || (e.shiftKey && key === "Tab"))) {
-          this.closeCalciteDropdown();
-        }
-        return;
-      }
-      switch (key) {
-        case " ":
-        case "Enter":
-          this.openCalciteDropdown();
-          break;
-        case "Escape":
-          this.closeCalciteDropdown();
-          break;
-      }
+
+    if (this.active && (key === "Escape" || (e.shiftKey && key === "Tab"))) {
+      this.closeCalciteDropdown();
+      return;
+    }
+
+    switch (key) {
+      case " ":
```

Comment:
Sidebar: this could use the `isActionKey` util if we switch (ba dum tss 🥁 ) to an `if`/`else` statement.